website/static/js/contribManager.js

```diff
@@ -6,106 +6,127 @@ var bootbox = require('bootbox');
 // out the permissions-related functions and remove currentUserCanEdit.
-var ContributorModel = function(contributor, currentUserCanEdit, pageOwner, isRegistration, isAdmin) {
+var ContributorModel = function(contributor, currentUserCanEdit, pageOwner, isRegistration, isAdmin, index, parent) {
 
     var self = this;
     $.extend(self, contributor);
 
-    self.permissionList = [
-        {value: 'read', text: 'Read'},
-        {value: 'write', text: 'Read + Write'},
-        {value: 'admin', text: 'Administrator'}
-    ];
-    self.getPermission = function(permission) {
-        for(var i = 0; i < self.permissionList.length; i++) {
-            if(permission === self.permissionList[i].value) {
-                return self.permissionList[i];
+    self.originals = {
+        permission: contributor.permission,
+        visible: contributor.visible,
+        index: index
+    };
+
+    self.permission = ko.observable(contributor.permission);
+
+    self.permissionText = ko.observable(parent.permissionDict[self.permission()]);
+
+    self.visible = ko.observable(contributor.visible);
+
+    self.permission.subscribeChanged(function(newValue, oldValue) {
+        if (oldValue === 'admin') {
+            parent.adminCount(parent.adminCount() - 1);
+        }
+        if (newValue === 'admin') {
+            parent.adminCount(parent.adminCount() + 1);
+        }
+        self.permissionText(parent.permissionDict[newValue]);
+    });
+
+    self.visible.subscribeChanged(function(newValue, oldValue) {
+        if (oldValue === true) {
+            parent.visibleCount(parent.visibleCount() - 1);
+        }
+        if (newValue === true) {
+            parent.visibleCount(parent.visibleCount() + 1);
+        }
+    });
+
+    self.permissionChange = ko.computed(function() {
+        return self.permission() !== self.originals.permission;
+    });
+
+    self.reset = function() {
+        if (self.deleteStaged()) {
+            if (self.visible()) {
+                parent.visibleCount(parent.visibleCount() + 1);
             }
+            if (self.permission() === 'admin') {
+                parent.adminCount(parent.adminCount() + 1);
+            }
+            self.deleteStaged(false);
         }
-        return self.permissionList[0];
+        self.permission(self.originals.permission);
+        self.visible(self.originals.visible);
+        self.index(self.originals.index);
     };
 
+    self.index = ko.observable(index);
+
     self.currentUserCanEdit = currentUserCanEdit;
     self.isAdmin = isAdmin;
-    self.visible = ko.observable(contributor.visible);
-    self.permission = ko.observable(contributor.permission);
-    self.curPermission = ko.observable(self.getPermission(self.permission()));
-    self.deleteStaged = ko.observable(contributor.deleteStaged || false);
-    self.removeContributor = 'Remove contributor';
+
+    self.deleteStaged = ko.observable(false);
```

Comment:
@sloria When contributors are passed into this viewModel there is no attribute for `deleteStaged` therefore it will always evaluate to false, so I just went ahead and made it go straight to false.  `deleteStaged` is just used to determine whether the user has staged a delete on a contributor but not saved this change yet.  If they save, the user is removed, if they navigate away and don't save, the changes are lost and therefore the delete should not be staged when they navigate back to the page.  I had a bit of trouble typing that out, so if you have questions feel free to ping me or come find me.